File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    tmp=0
    for i in range(left,right+1):
        lst=[]
        for j in range(1,i+1):
            if i % j == 0:
                lst.append(j)
        if len(lst)%2==0:
            tmp += i
        else:
            tmp-= i
    return tmp

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    private int answer = 0;
    
    public int solution(int left, int right) {
        for(int i=left; i<=right; i++){
            cal(i);
        }
        return answer;
    }
    
    public void cal(int n){
        int tmp = 1;
        for(int i=2; i<=n; i++){
            if(n%i==0){
                ++tmp;
            }
        }
        answer += tmp%2==0? n: -n;
    }
}